src/renderer/prefComponents/image/components/uploader/index.vue

```diff
@@ -0,0 +1,267 @@
+        <div class="form-group">
+          <div class="label">
+            GitHub token:
+            <el-tooltip class="item" effect="dark"
+              content="The token is saved by Keychain on macOS, Secret Service API/libsecret on Linux and Credential Vault on Windows"
+              placement="top-start">
+              <i class="el-icon-info"></i>
+            </el-tooltip>
+          </div>
+          <el-input v-model="githubToken" placeholder="Input token" size="mini"></el-input>
+        </div>
+        <div class="form-group">
+          <div class="label">Owner name:</div>
+          <el-input v-model="github.owner" placeholder="owner" size="mini"></el-input>
+        </div>
+        <div class="form-group">
+          <div class="label">Repo name:</div>
+          <el-input v-model="github.repo" placeholder="repo" size="mini"></el-input>
+        </div>
+        <div class="form-group">
+          <div class="label">Branch name (optional):</div>
+          <el-input v-model="github.branch" placeholder="branch" size="mini"></el-input>
+        </div>
+        <legal-notices-checkbox class="github"
+          :class="[{ 'error': legalNoticesErrorStates.github }]"
+          :uploaderService="uploadServices.github"></legal-notices-checkbox>
+        <div class="form-group">
+          <el-button size="mini" :disabled="githubDisable" @click="save('github')">Save
+          </el-button>
+        </div>
+      </div>
+      <div class="script" v-else-if="currentUploader === 'cliScript'">
+        <div class="description">The script will be executed with the image file path as its only
+          argument and it should output any valid value for the <code>src</code> attribute of a
+          <em>HTMLImageElement</em>.
+        </div>
+        <div class="form-group">
+          <div class="label">Shell script location:</div>
+          <el-input v-model="cliScript" placeholder="Script absolute path" size="mini"></el-input>
+        </div>
+        <div class="form-group">
+          <el-button size="mini" :disabled="cliScriptDisable" @click="save('cliScript')">Save
+          </el-button>
+        </div>
+      </div>
+    </section>
+  </div>
+</template>
+
+<script>
+import { shell } from 'electron'
+import services, { isValidService } from './services.js'
+import legalNoticesCheckbox from './legalNoticesCheckbox'
+import { isFileExecutableSync } from '@/util/fileSystem'
+import CurSelect from '@/prefComponents/common/select'
+import commandExists from 'command-exists'
+import notice from '@/services/notification'
+
+export default {
+  components: {
+    legalNoticesCheckbox,
+    CurSelect
+  },
+  data () {
+    this.uploaderOptions = Object.keys(services).map(name => {
+      const { name: label } = services[name]
+      return {
+        label,
+        value: name
+      }
+    })
+    return {
+      githubToken: '',
+      github: {
```

Comment:
We could remove all deprecated modes and introduce this as breaking change. Now, you know that you have to remove and what not. I think there's no point in waiting for the next version?

Reply:
>We could remove all deprecated modes and introduce this as breaking change. Now, you know that you have to remove and what not. I think there's no point in waiting for the next version?

Mainly considering that there are still users using GitHub as an image upload service, leaving time for these users to migrate to picgo, if they are removed directly, GitHub uploader will not work.

Reply:
You have to decide whether this is important or not because users are forced to do this in the next release.

Reply:
ok